src/implementation.ts

```diff
@@ -249,6 +249,10 @@ function getBatchLoader(resourceConfig: BatchResourceConfig, resourcePath: Reado
                                         )} Resource returned \${response.length} items, but we requested \${requests.length} items.\`,
                                         'Add reorderResultsByKey to the config for this resource to be able to handle a partial response.',
                                     ].join(' '));
+
+                                    // Tell flow that "response" is actually an error object.
+                                    // (This is so we can pass it as 'cause' to CaughtResourceError)
+                                    invariant(response instanceof Error, 'expected BatchItemNotFoundError to be an Error');
```

Comment:
Hmm why is this needed? We made response an error in the line just above

Here's a similar repro:

https://flow.org/try/#0PTAEDMBsHsHcCh4GNIEMDO7QDFrQKIBOh0hoApgB4Au5AdgCZZElkDeAvgNzLR3rUIeUAF5QdcrBx4WpABQByAJ4BLANbl0CgJQ94ICDCnpy5ALZYAFqgAONpaFgrql0C5XoAXPDng8n0FlCbURwAFc6JGoVPlBrRkhyILlyYgCg7VA2eABIAwZoUFQ3SxU6AHMAOmrQeA5EA1RGEo9veIZE5L9oXSA

Is just this a flow version thing? Or can you show a similar minimal repro that errors with our current logic?

Reply:
Root cause comes from https://unpkg.com/dataloader-codegen@0.1.4/lib/runtimeHelpers.js.flow and search `BatchItemNotFoundError`
you can see `declare export class BatchItemNotFoundError mixins Error`, this is not correct, we should use `extends` rather than `mixins`

Reply:
I think the way we generate Flow from TS using `flowgen` is not perfect https://github.com/joarwilk/flowgen/blob/80792a4f21c5e5707443a9d82550493b47c14885/src/printers/declarations.js#L299

Reply:
ah I see. Ok makes sense, thanks for doing the digging here!

Can we add a comment about _why_ we're adding the invariant, linking to this conversation or something? Otherwise it looks redundant when reading the code.

(Bonus points if we can make a minimal repro and make a ticket on the flowgen project - not necessary tho)

After the comment I think we're good to ship!

Reply:
Minimal repro is https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVATApgYxgQwCcsw98BncsAMTjgFFDC5CwBbASwA8OA7Kxs1YBvAL6oAFLQZMWALjCCWASiA, I will create an issue on the flowgen project.

yea, I will add some comments about the reason why I add redundant code here.